### ISSUE TITLE
no-unnecessary-class: false positive with decorated classes

### ISSUE THREAD

**Issue by unlight** [Microsoft/tslint-microsoft-contrib#494]
Very common case in `nestjs` framework.
 
```ts
@Module({
    controllers: [HelloController],
    providers: [],
})
export class HelloModule {}
```

**Comment by JoshuaKGoldberg** [comment]
Indeed! It'd be great to have a feature on the rule to allow decorated classes.

**Comment by IllusionMH** [comment]
As I can see - `no-unnecessary-class` is part of core rules. (`no-stateless-class` was deprecated in favor of core rule)

@JoshuaKGoldberg maybe it make sense to create feature request in [palantir/tslint](https://github.com/palantir/tslint) repo? I can do it if needed.

To avoid error in provided case - it is possible to use `"allow-empty-class"` option, to avoid error is this case (but no error for classes with decorators)

**Comment by JoshuaKGoldberg** [comment]
Also see https://github.com/palantir/tslint/issues/1142
